docs/reference/analysis/tokenfilters/flatten-graph-tokenfilter.asciidoc

```diff
@@ -4,18 +4,232 @@
 <titleabbrev>Flatten graph</titleabbrev>
 ++++
 
-The `flatten_graph` token filter accepts an arbitrary graph token
-stream, such as that produced by
-<<analysis-synonym-graph-tokenfilter>>, and flattens it into a single
-linear chain of tokens suitable for indexing.
-
-This is a lossy process, as separate side paths are squashed on top of
-one another, but it is necessary if you use a graph token stream
-during indexing because a Lucene index cannot currently represent a
-graph.  For this reason, it's best to apply graph analyzers only at
-search time because that preserves the full graph structure and gives
-correct matches for proximity queries.
-
-For more information on this topic and its various complexities,
-please read the http://blog.mikemccandless.com/2012/04/lucenes-tokenstreams-are-actually.html[Lucene's
-TokenStreams are actually graphs] blog post.
+Flattens a <<token-graphs,token graph>> produced by a graph token filter, such
+as <<analysis-synonym-graph-tokenfilter,`synonym_graph`>> or
+<<analysis-word-delimiter-graph-tokenfilter,`word_delimiter_graph`>>.
+
+Flattening a token graph containing
+<<token-graphs-multi-position-tokens,multi-position tokens>> makes the graph
+suitable for <<analysis-index-search-time,indexing>>. Otherwise, indexing does
+not support token graphs containing multi-position tokens.
+
+[WARNING]
+====
+Flattening graphs containing multi-position tokens is a lossy process.
+Flattening collapses the side paths of a token graph into a linear path.
+
+If possible, avoid using graph token filters in
+<<analysis-index-search-time,index analyzers>>. Use the respective
+<<analysis-synonym-tokenfilter,`synonym`>> or
+<<analysis-word-delimiter-tokenfilter,`word_delimiter`>> filters instead.
```

Comment:
I think it would be good to reword this to strongly advise against using these at index time at all - generally speaking, filters that could produce graphs should be used at query time instead.  `flatten_graph` is a kind of shonky workaround for a problem that really should be fixed elsewhere.

Reply:
Thanks for the feedback @romseygeek!

With afcbebfb5ea49d3fe008d371fedd3ffb3e85cc0e, I reworded the warning to remove the recommendation for the `synonym` and `word_delimiter` filters. It also recommends that token graph filters only be used in search analyzers.

Let me know if you see any other needed changes!